### PR TITLE
Adds flatten helper

### PR DESCRIPTION
**Pull request by vikram7**
This PR adds the `flatten`helper per https://github.com/DockYard/ember-composable-helpers/issues/32

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,21 +71,22 @@ For action helpers, this will mean better currying semantics:
   + [`underscore`](#underscore)
   + [`w`](#w)
 * [Array](#array-helpers)
-  + [`map-by`](#map-by)
```
Comment (poteto) [inline README.md]: Please don't sort, you can do it as a separate PR.

File: addon/helpers/flatten.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import Ember from 'ember';
+
+const {
+  Helper: { helper }
+} = Ember;
+
+export function flatten(array) {
+  array = array || [];
+
+  return array.reduce((flat, toFlatten) => {
```
Comment (poteto) [inline addon/helpers/flatten.js]: This helper will not recompute when an array has changes. Please see something like the `contains` helper for an example.

File: tests/integration/helpers/flatten-test.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+test('it handles null input', function(assert) {
+  this.render(hbs`
+    {{#each (flatten null) as |number|}}{{number}}{{/each}}
+  `);
+
+  assert.equal(this.$().text().trim(), '', 'should handle null input');
+});
+
+test('it handles undefined input', function(assert) {
```
Comment (poteto) [inline tests/integration/helpers/flatten-test.js]: Needs another test for array changes

File: addon/helpers/flatten.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import Ember from 'ember';
+
+const {
+  Helper: { helper }
+} = Ember;
+
+export function flatten(array) {
+  array = array || [];
```
Comment (poteto) [inline addon/helpers/flatten.js]: Can we have this as `array || [array];`? I would prefer that this method always returns an array regardless of the value passed into it.

Also, we should be using an `isArray` check here. It is still valid if the array contains `null` or `undefined` values.

**Comment by poteto** [comment]
If you rebase this against master, `beta` will be an allowed failure.

File: tests/integration/helpers/flatten-test.js (hunk 3 above)
Comment (vikram7) [inline tests/integration/helpers/flatten-test.js]: :+1: Has been added above (line 25).

File: addon/helpers/flatten.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  isArray,
+  observer,
+  set
+} = Ember;
+
+export function flatten(array) {
+  array = array || [array];
+
+  return array.reduce((flat, toFlatten) => {
+    return flat.concat(Array.isArray(toFlatten) ? flatten(toFlatten) : toFlatten);
```
Comment (poteto) [inline addon/helpers/flatten.js]: Can we use `Ember.typeOf` here? Just so that other kinds of Ember Arrays will also be true here.

**Comment by taras** [comment]
@vikram7 @poteto what happened to this PR? Should we re-born it?

**Comment by martndemus** [comment]
@taras this got stale and apparently the OP closed it. Feel free to open a fresh PR!
